samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemoryExtractor.cs

```diff
@@ -22,42 +22,49 @@ internal static class SemanticChatMemoryExtractor
     /// Returns the name of the semantic text memory collection that stores chat semantic memory.
     /// </summary>
     /// <param name="chatId">Chat ID that is persistent and unique for the chat session.</param>
-    /// <param name="memoryName">Name of the memory category</param>
-    internal static string MemoryCollectionName(string chatId, string memoryName) => $"{chatId}-{memoryName}";
+    /// <param name="memoryType">Name of the memory category</param>
+    internal static string MemoryCollectionType(string chatId, string memoryType) => $"{chatId}-{memoryType}";
 
     /// <summary>
     /// Extract and save semantic memory.
     /// </summary>
     /// <param name="chatId">The Chat ID.</param>
-    /// <param name="kernel">The semantic kernel.</param>
-    /// <param name="context">The context containing the memory.</param>
+    /// <param name="context">The SKContext</param>
     /// <param name="options">The prompts options.</param>
+    /// <param name="chatPlugin">The plugin containing chat specific semantic functions as prompt templates.</param>
+    /// <param name="chatPromptOptions">The token counts of the prompt text templates in the chatPlugin.</param>
     internal static async Task ExtractSemanticChatMemoryAsync(
         string chatId,
-        IKernel kernel,
         SKContext context,
-        PromptsOptions options)
+        PromptsOptions options,
+        IDictionary<string, ISKFunction> chatPlugin,
```

Comment:
nit: probably worth defining type aliases for these types

Reply:
We dont alias IDictionary<string, ISKFunction> in the core kernel, and we dont use it in enough places; Aliasing it imo makes it harder to read; Similarly IDictionary<string, PluginPromptOptions> is only used in 2 locations.